### PR TITLE
Feature/playlist handling. Also final music pr 🎉🎉🎉🎉

### PR DESCRIPTION
**Pull request by weeryan17**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Added/Changed
Added save and load commands for queue.

### CODE REVIEW

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            scope = EnumUtils.getEnum(PlaylistType.class, context.getArg(1).toUpperCase());
+            if (scope == null) {
+                context.getTypedMessaging().replyDanger("Scope `" + context.getArg(1) + "` not found");
+                return;
+            }
+        }
+
+        long owner = 0;
+        switch (scope) {
+
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java]: Blank line

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            case USER:
+                owner = sender.getUser().getIdLong();
+                break;
+        }
+
+        long lambdaOwner = owner;
+        PlaylistType lambdaScope = scope;
+        CascadePlayer.SavePlaylistResult result = context.getMusicPlayer().saveCurrentPlaylist(lambdaOwner, lambdaScope, context.getArg(0), false);
+        switch (result) {
+
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java]: blank line

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                        context.getTypedMessaging().replyWarning("Playlist already exists in guild and you don't have the perm `cascade.queue.dave.overwrite` to overwrite it."); //TODO actually get the perm
+                        return;
+                    }
+                }
+                ConfirmUtils.confirmAction(sender.getUser().getIdLong(), "overwrite", context.getChannel(), MessageType.WARNING,
+                        "Playlist already exists. Would you like to overwrite it?", TimeUnit.SECONDS.toMillis(1), TimeUnit.SECONDS.toMillis(10), new ConfirmUtils.ConfirmRunnable() {
+                            @Override
+                            public void execute() {
+                                context.getMusicPlayer().saveCurrentPlaylist(lambdaOwner, lambdaScope, context.getArg(0), false);
+                                context.getTypedMessaging().replySuccess("Saved playlist `" + context.getArg(0) + "`");
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java]: Maybe say what scope the playlist gets stored in?

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                            }
+                        });
+                break;
+            case NEW:
+                context.getTypedMessaging().replySuccess("Saved playlist `" + context.getArg(0) + "`");
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueSaveSubCommand.java]: Same ^

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.cascadebot.cascadebot.utils.buttons.Button;
+import org.cascadebot.cascadebot.utils.buttons.ButtonGroup;
+
+import java.util.Set;
+
+public class QueueLoadSubCommand implements ICommandExecutable {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        context.getMusicPlayer().loadPlaylist(context.getArg(0), sender, (result, tracks) -> {
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java]: if arg 0 doesn't exist this is going to throw an NPE

File: src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java [hunk 6]
```diff
@@ -34,4 +34,7 @@
     public static final String FROWNING = "\uD83D\uDE26"; // 😦
     public static final String THINKING = "\uD83E\uDD14"; // 🤔
 
+    public static final String ONE = "\uD83C\udde6";
+    public static final String TWO = "\uD83C\udde7";
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java]: I'm gonna go and say add all the numbers. Also, `\uD83C\udde6` is not `1`? It's 🇦

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return CascadePermission.of("Queue load sub command", "queue.load", true);
+    }
+
+    @Override
+    public String description() {
+        return null;
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java]: Can you make this with @Deprecated(forRemoval=true) so I know to remove it for lang?

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                        }
+                        message.delete().queue();
+                        context.getMusicPlayer().loadPlaylist(context.getArg(0), sender, PlaylistType.GUILD, ((loadPlaylistResult, newTracks) -> {
+                            context.getUIMessaging().sendTracksFound(newTracks);
+                        }));
+                    })));
+                    context.getUIMessaging().sendButtonedMessage("Where you like to load this track from\n" + UnicodeConstants.ONE +
+                            " User\n" + UnicodeConstants.TWO + " Guild", buttonGroup);
+                    break;
+                case DOESNT_EXISTS:
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/queue/QueueLoadSubCommand.java]: Should be `DOESNT_EXIST`

File: src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java (hunk 6 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java]: Whops

**Pull request merged**